client/job.lua

```diff
@@ -410,139 +399,109 @@ if config.useTarget then
 else
     CreateThread(function()
         for i = 1, #sharedConfig.locations.duty do
-            local function enteredSignInZone()
-                if not QBX.PlayerData.job.onduty then
-                    lib.showTextUI(locale('text.onduty_button'))
-                else
-                    lib.showTextUI(locale('text.offduty_button'))
-                end
-            end
-
-            local function outSignInZone()
-                lib.hideTextUI()
-            end
-
-            local function insideDutyZone()
-                OnKeyPress(toggleDuty)
-            end
-
             lib.zones.box({
                 coords = sharedConfig.locations.duty[i],
                 size = vec3(1, 1, 2),
                 rotation = -20,
                 debug = config.debugPoly,
-                onEnter = enteredSignInZone,
-                onExit = outSignInZone,
-                inside = insideDutyZone,
+                onEnter = function()
+                    if not QBX.PlayerData.job.onduty then
+                        lib.showTextUI(locale('text.onduty_button'))
+                    else
+                        lib.showTextUI(locale('text.offduty_button'))
+                    end
+                end,
```

Comment:
```suggestion
                onEnter = function()
                    local label = QBX.PlayerData.job.onduty and 'text.offduty_button' or 'text.onduty_button'
                    lib.showTextUI(locale(label))
                end,
```
Depending on readability preference this could be stuffed inside the locale call